README.md

```diff
@@ -47,24 +47,38 @@ use {
         -- to trigger the close_events of vim.diagnostic.open_float.
         function ()
           -- Close floating diag. and make it triggerable again.
-          vim.cmd('doautocmd InsertEnter')
+          vim.api.nvim_exec_autocmds("User", { pattern = "MurmurDiagnostics" })
           vim.w.diag_shown = false
         end,
       }
     }
 
     -- To create IDE-like no blinking diagnostic message with `cursor` scope. (should be paired with the callback above)
-    vim.api.nvim_create_autocmd({ 'CursorHold' }, {
+    vim.api.nvim_create_autocmd({ 'CursorHold', 'InsertLeave' }, {
       group = FOO,
       pattern = '*',
       callback = function ()
         -- skip when a float-win already exists.
         if vim.w.diag_shown then return end
 
         -- open float-win when hovering on a cursor-word.
-        if vim.w.cursor_word ~= '' then
-          vim.diagnostic.open_float()
+        if vim.w.cursor_word ~= "" then
+          local buf = vim.diagnostic.open_float({
+            scope = "cursor",
+            -- Only close the window on InsertEnter and the explicit diagnostic close event
+            close_events = { "InsertEnter", "User MurmurDiagnostics" },
```

Comment:
TIL. I tried very hard to know how to trigger my user event but gave up.

Reply:
Yep! Learned this pretty recently as well haha